src/coreComponents/constitutive/solid/DelftEgg.hpp

```diff
@@ -0,0 +1,618 @@
+
+    // real64 h = 1.0 / (Cc-Cr); //Linear hardening version
+    pc = oldPc * std::exp( -1./(Cc-Cr)*(eps_v_trial-solution[0]));
+    // pc = oldPc + h *(eps_v_trial-solution[0]); //Linear hardening version
+
+    evaluateYield( trialP, trialQ, pc, M, alpha, Cc, Cr, bulkModulus, mu, yield, df_dp, df_dq, df_dpc, df_dp_dve, df_dq_dse );
+    real64 dpc_dve = -1./(Cc-Cr) * pc;
+
+    real64 scale = 1./(mu*mu);
+    // assemble residual system
+    residual[0] = solution[0] - eps_v_trial + solution[2]*df_dp;   // strainElasticDev - strainElasticTrialDev + dlambda*dG/dPQ = 0
+    residual[1] = solution[1] - eps_s_trial + solution[2]*df_dq;         // strainElasticVol - strainElasticTrialVol + dlambda*dG/dQ = 0
+    residual[2] = yield * scale;      // F = 0
+
+
+    // check for convergence
+
+    norm = LvArray::tensorOps::l2Norm< 3 >( residual );
+    if( iter==0 )
+    {
+      normZero = norm;
+      normOld = norm;
+    }
+
+    if( norm < 1e-12*(normZero+1.0))
+    {
+      break;
+
+    }
+    else if( iter > 0 && norm>normOld && cuts<maxCuts ) //linesearch
+    {
+      cuts++;
+      iter--;
+      delta[0] *= 0.5;
+      delta[1] *= 0.5;
+      delta[2] *= 0.5;
+      solution[0] += delta[0];
+      solution[1] += delta[1];
+      solution[2] += delta[2];
+      normOld = norm;
+    }
+    else
+    {
+      // solve Newton system
+      normOld = norm;
+      cuts=0;
+      real64 dp_dve = bulkModulus;
+      real64 dq_dse = 3. *mu;
+
+      jacobian[0][0] = 1. + solution[2] * df_dp_dve;
+      jacobian[0][2] = df_dp;
+      jacobian[1][1] = 1. + solution[2]*df_dq_dse;
+      jacobian[1][2] = df_dq;
+      jacobian[2][0] = (dp_dve * df_dp - dpc_dve * df_dpc)*scale;
+      jacobian[2][1] = (dq_dse * df_dq)*scale;
+      jacobian[2][2] = 0.0;
+
+
+      LvArray::tensorOps::invert< 3 >( jacobianInv, jacobian );
+      LvArray::tensorOps::Ri_eq_AijBj< 3, 3 >( delta, jacobianInv, residual );
+
+      for( localIndex i=0; i<3; ++i )
+      {
+        solution[i] -= delta[i];
+      }
+    }
+  }
+
+
+  // re-construct stress = P*eye + sqrt(2/3)*Q*nhat
+
+  twoInvariant::stressRecomposition( trialP,
+                                     trialQ,
+                                     deviator,
+                                     stress );
+
+  // construct consistent tangent stiffness
+
+  LvArray::tensorOps::fill< 6, 6 >( stiffness, 0.0 );
+  real64 BB[2][2] = {{}};
+
+  //  real64 dpc_dve = 1./(Cc-Cr);//-1./(Cc-Cr) * pc; //linear hardening version
+  real64 const c = alpha/(alpha+1.)*pc;
+  real64 dpc_dve = -1./(Cc-Cr) * pc;
+  real64 df_dp_depsv;
+  real64 factor;
+  if( trialP >= c )   // Use MCC
+  {
+
+    factor = 2.*alpha/ (alpha+1.);
+    df_dpc = -factor * trialP;
+    df_dp_depsv = factor * dpc_dve;
+  }
+  else
+  {
+    factor = 2. * alpha*alpha*alpha / (alpha+1.);
+    df_dpc = 2. * alpha*alpha*(alpha-1.) /(alpha+1.) * pc - factor * trialP;
+    df_dp_depsv = factor * dpc_dve;
+  }
+
+
+  real64 a1= 1. + solution[2]*df_dp_depsv;
```

Comment:
always add space after = (ditto through this whole function)
```suggestion
  real64 a1 = 1. + solution[2]*df_dp_depsv;
```

Reply:
done